x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/relevance_tuning.test.tsx

```diff
@@ -50,43 +50,48 @@ describe('RelevanceTuning', () => {
 
   it('renders', () => {
     const wrapper = subject();
+    expect(wrapper.find(RelevanceTuningCallouts).exists()).toBe(true);
     expect(wrapper.find(RelevanceTuningForm).exists()).toBe(true);
-    expect(wrapper.find(Loading).exists()).toBe(false);
-    expect(wrapper.find(EmptyState).exists()).toBe(false);
+    expect(wrapper.find(RelevanceTuningPreview).exists()).toBe(true);
   });
 
   it('initializes relevance tuning data', () => {
     subject();
     expect(actions.initializeRelevanceTuning).toHaveBeenCalled();
   });
 
-  it('will render an empty message when the engine has no schema', () => {
+  it('will prevent user from leaving the page if there are unsaved changes', () => {
     setMockValues({
       ...values,
-      engineHasSchemaFields: false,
+      unsavedChanges: true,
     });
-    const wrapper = subject();
-    expect(wrapper.find(EmptyState).dive().find(EuiEmptyPrompt).exists()).toBe(true);
-    expect(wrapper.find(Loading).exists()).toBe(false);
-    expect(wrapper.find(RelevanceTuningForm).exists()).toBe(false);
+    expect(subject().find(UnsavedChangesPrompt).prop('hasUnsavedChanges')).toBe(true);
```

Comment:
UnsavedChangesPrompt could eventually go in the page template!

Reply:
Ooo!!! Nice idea, I love it!